certbot/certbot/_internal/plugins/webroot.py

```diff
@@ -157,7 +157,8 @@ def _create_challenge_dirs(self):
                 "--webroot-path and --domains, or --webroot-map. Run with "
                 " --help webroot for examples.")
         for name, path in path_map.items():
-            self.full_roots[name] = os.path.join(path, challenges.HTTP01.URI_ROOT_PATH)
+            self.full_roots[name] = os.path.join(path, os.path.normcase(
+                challenges.HTTP01.URI_ROOT_PATH))
```

Comment:
How challenging do you think it would be to fix `URI_ROOT_PATH` itself? It's a little absurd that we hard-code the `/` in there in the first place.

Reply:
I do not think that `URI_ROOT_PATH` needs to be fixed in fact. Even though it is interpretated as a filesystem path for the webroot plugin to know where to put the challenge file, it's primary usage is to be an URI fragment that constructs the URL of a HTTP-01 challenge.

In that extent this representation is independent from an OS, and `/` is a valid separator for URIs.

Reply:
Ah, good point, I guess it sure was right there in the name.